### PR TITLE
[drape] Fixed wrong line height for NotoSansCJK on Android

### LINKED ISSUE #8708: Incorrect text layout for CJK fonts
The line height is wrongly calculated for texts in Chinese for NotoSansCJK font, leading to wrongly layouted text lines with these characters (they are shifted down too much). The issue does not reproduce with non-CJK fonts:

telegram-cloud-photo-size-1-4945168329379785905-w
telegram-cloud-photo-size-1-4945168329379785906-w
telegram-cloud-photo-size-1-4945168329379785907-w
telegram-cloud-photo-size-1-4945168329379785910-y

telegram-cloud-photo-size-1-4945168329379785908-w

### PR DESCRIPTION
**Pull request by biodranik**
This system font (found and used by OM on the start) reports too big line height (normal bbox.yMax, and too big negative bbox.yMin).

The previous implementation could wrongly mix vertical advance with bbox.yMin calculation that leads to way too high line height (e.g. 55px when 22px have been specified). This caused wrongly layouted text in some cases.

Fixes #8708

выява

### CODE REVIEW

File: drape/glyph_manager.cpp [hunk 1]
```diff
@@ -258,16 +258,22 @@ class Font
...
-  int32_t GetLineHeigthPixels() const
+private:
+  int32_t GetLineHeigthPixels(FT_Pos vertAdvance) const
   {
-    auto const vertAdvance = m_fontFace->glyph->metrics.vertAdvance;
-    auto const yMax = vertAdvance && vertAdvance < m_fontFace->bbox.yMax ? vertAdvance : m_fontFace->bbox.yMax;
-    int32_t const bboxYMax = FT_MulFix(yMax, m_fontFace->size->metrics.y_scale) >> 6;
+    // Freetype says that vertAdvance is unreliable if it contains a synthesized value in case of FT_HAS_VERTICAL
+    // returning false. Testing on supported fonts shows that non-zero vertAdvance can be trusted as a line height.
+    if (vertAdvance && vertAdvance < m_fontFace->bbox.yMax) [[likely]]
```
Comment (vng) [inline drape/glyph_manager.cpp:271]: Expicitly vertAdvance != 0 (or probably vertAdvance > 0, IDK)
[[likely]] :)

**Review by vng — APPROVED** [review]

File: drape/glyph_manager.cpp (hunk 1 above)
Comment (biodranik) [reply drape/glyph_manager.cpp]: Thanks, fixed. 

What's funny about [[likely]]? It's called hundreds of times per second.

Comment (vng) [reply drape/glyph_manager.cpp]: I don't think that [likely] can change something in this code and expect to have some reasonable tradeoff between these attributes and a clean code. Up to you here.

Comment (biodranik) [reply drape/glyph_manager.cpp]: It's easy to check without guessing. There is a difference in the generated code with and without likely with armv7 and x86_64 archs: https://godbolt.org/z/s5Tfs1jWx

Comment (biodranik) [reply drape/glyph_manager.cpp]: Google actively uses these attributes in its code for a reason.

**Pull request merged**